packages/cli/init/src/utils/loadOpenApiFromUrl.ts

```diff
@@ -44,3 +42,18 @@ export async function loadOpenAPIFromUrl({ url, logger }: { url: string; logger:
         };
     }
 }
+
+async function fetchOpenAPIFromUrl({ url, logger }: { url: string; logger: Logger }): Promise<string> {
+    const response = await axios.get(url);
+    const contentType = response.headers["content-type"] ?? "";
+    if (contentType.includes("json")) {
+        return dump(response.data);
+    }
+    if (contentType.includes("yaml")) {
+        return response.data;
+    }
+    logger.warn(
+        `Unrecognized Content-Type "${contentType}" from endpoint ${url}. Please ensure you're pointing to a URL that returns JSON or YAML and not HTML (e.g. Swagger UI webpage)`
+    );
+    return response.data;
```

Comment:
Accidentally missed this from my last review, but do you mind preserving the original behavior in the default case? We can assume we've received JSON and dump it to YAML like so:

```suggestion
    return dump(response.data);
```